tests/e2e/specs/upe-split/shopper/shopper-deferred-intent-creation-upe-enabled.spec.js

```diff
@@ -102,6 +102,39 @@ describe( 'Enabled UPE with deferred intent creation', () => {
 			await shopperWCP.deleteSavedPaymentMethod( card.label );
 			await expect( page ).toMatch( 'Payment method deleted' );
 		} );
+
+		it( 'should show terms when checking the save checkbox', async () => {
+			await setupProductCheckout(
+				config.get( 'addresses.customer.billing' )
+			);
+			await fillCardDetails( page, card );
+			await shopperWCP.toggleSavePaymentMethod();
+
+			const frameHandle = await page.waitForSelector(
+				'#payment .payment_method_woocommerce_payments .wcpay-upe-element iframe'
+			);
+			const stripeFrame = await frameHandle.contentFrame();
+
+			const textElement = await stripeFrame.$( '.TermsText' );
+
+			const text = textElement.evaluate( ( el ) => el.innerText );
+
+			expect( text ).toContain(
+				'By providing your card information, you allow'
+			);
+		} );
+
+		it( 'should not allow guest user to save the card', async () => {
+			await shopperWCP.logout();
+			await setupProductCheckout(
+				config.get( 'addresses.customer.billing' )
+			);
+			await fillCardDetails( page, card );
+
+			await expect( page ).not.toHaveSelector(
```

Comment:
I don't think `toHaveSelector()` is actually a function. 
```suggestion
			await expect( page ).not.toMatchElement(
				'#wc-woocommerce_payments-new-payment-method'
			);
```